src/components/shared/Sidebar/Sidebar.component.tsx

```diff
@@ -0,0 +1,214 @@
+/**
+ * Sidebar Component
+ *
+ * Sidebar component for the application including menus and submenu.
+ */
+
+import React, { useCallback, useMemo, useState } from 'react';
+
+import { Link, Location, useLocation } from 'react-router';
+
+import {
+    ExpandLess,
+    ExpandMore,
+    Public,
+    SettingsOutlined,
+    Tune,
+} from '@mui/icons-material';
+import {
+    Box as MuiBox,
+    Collapse as MuiCollapse,
+    Divider as MuiDivider,
+    IconButton as MuiIconButton,
+    List as MuiList,
+    Toolbar as MuiToolbar,
+} from '@mui/material';
+
+import { StyledBadge } from '@components/common/Badge';
+import {
+    StyledBottomStack,
+    StyledDrawer,
+    StyledListItem,
+    StyledListItemIcon,
+    StyledListItemText,
+    StyledSubmenuList,
+} from '@components/shared/Sidebar/Sidebar.styles.tsx';
+import { SidebarProps } from '@components/shared/Sidebar/Sidebar.types.ts';
+import { SidebarItem, SidebarSection } from '@containers/Sidebar/config';
+import { theme } from '@theme';
+
+/**
+ * Renders a submenu for sidebar items.
+ */
+const renderSubmenu = (
+    submenu: SidebarItem[],
+    location: Location,
+    onItemClick: (sidebarItem: SidebarItem) => void,
+): React.ReactNode => (
+    <StyledSubmenuList disablePadding>
+        {submenu.map((subItem) => {
```

Comment:
by virtue of pure components, this component should never have access to a "list" of values. Creation of the list is the parent component's job.

renderList = (list) => list.map(listItem => <JSX {...listItem} />) // not standard

const RenderList = (props): React.FC => <JSX>{props.list.map(listItem => <JSX {...listItem} />)}</JSX> // not standard

const Item = (props): React.FC => <JSX /> // standard React
...
// parent
list.map(listItem => <Item {...listItem} />)

Reply:
Must be fixed in all places